managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3633,12 +3633,19 @@ public boolean isValidPosition(PositionImpl position) {
             log.debug("IsValid position: {} -- last: {}", position, last);
         }
 
-        if (position.getEntryId() < 0) {
+        if (!ledgers.containsKey(position.getLedgerId())){
             return false;
-        } else if (position.getLedgerId() > last.getLedgerId()) {
+        } else if (position.getEntryId() < 0) {
             return false;
+        } else if (position.getLedgerId() > last.getLedgerId() && position.getLedgerId() != currentLedger.getId()) {
+            return false;
```

Comment:
We shouldn't allow users to set to a position that after the last add confirmed entry. Because the add operation might get failed if the message not able to write to bookies.

Reply:
The answer is [below](https://github.com/apache/pulsar/pull/21248#discussion_r1337131858)